thunx-autoconfigure/src/test/java/com/contentgrid/thunx/api/autoconfigure/AbacAutoConfigurationTest.java

```diff
@@ -18,23 +18,71 @@ public class AbacAutoConfigurationTest {
+                    assertThat(context).hasBean("abacFilterRegistration");
+                    assertThat(context).hasBean("interceptRepositoryRestMvcConfiguration");
+                    assertThat(context).hasBean("ensureQueryDslPredication");
+                    assertThat(context).hasBean("ensureAbacQueryDslResolverExist");
+                    assertThat(context).doesNotHaveBean("abacJwtAuthenticationConverter");
+                });
+    }
+
+    @Test
+    public void shouldEnableAbacWhenPropertyEqualsHeader() {
+
+        contextRunner.withUserConfiguration(TestContext.class)
+                .withSystemProperties("contentgrid.thunx.abac.source=header")
+                .run((context) -> {
+                    assertThat(context).hasSingleBean(QuerydslBindingsFactory.class);
+                    assertThat(context).hasSingleBean(ThunkExpressionDecoder.class);
+                    assertThat(context).hasSingleBean(AbacExceptionHandler.class);
+                    assertThat(context).hasSingleBean(AbacRequestFilter.class);
+                    assertThat(context).hasBean("abacFilterRegistration");
+                    assertThat(context).hasBean("interceptRepositoryRestMvcConfiguration");
+                    assertThat(context).hasBean("ensureQueryDslPredication");
+                    assertThat(context).hasBean("ensureAbacQueryDslResolverExist");
+                    assertThat(context).doesNotHaveBean("abacJwtAuthenticationConverter");
+                });
+    }
+
+    @Test
+    public void shouldDisableAbacWhenPropertyEqualsNone() {
+
+        contextRunner.withUserConfiguration(TestContext.class)
+                .withSystemProperties("contentgrid.thunx.abac.source=none")
+                .run((context) -> {
+                    assertThat(context).hasSingleBean(QuerydslBindingsFactory.class);
+                    assertThat(context).hasSingleBean(ThunkExpressionDecoder.class);
+                    assertThat(context).hasSingleBean(AbacExceptionHandler.class);
+                    assertThat(context).doesNotHaveBean(AbacRequestFilter.class);
+                    assertThat(context).doesNotHaveBean("abacFilterRegistration");
+                    assertThat(context).hasBean("interceptRepositoryRestMvcConfiguration");
+                    assertThat(context).hasBean("ensureQueryDslPredication");
+                    assertThat(context).hasBean("ensureAbacQueryDslResolverExist");
+                    assertThat(context).doesNotHaveBean("abacJwtAuthenticationConverter");
+                });
+    }
+
+    @Test
+    public void shouldFailWhenPropertyIsInvalid() {
+        contextRunner.withUserConfiguration(TestContext.class)
+                .withSystemProperties("contentgrid.thunx.abac.source=invalid")
+                .run(context -> {
+                    assertThat(context).hasFailed();
+                    assertThat(context).getFailure().isInstanceOf(IllegalArgumentException.class);
```

Comment:
This does not really test a lot.

Instead, either throw a specific exception and check for that, or check the exception message to be what you expect.